pallets/connectors/src/lib.rs

```diff
@@ -550,6 +642,26 @@ pub mod pallet {
 
 			encoded
 		}
+
+		/// Returns the `u128` general index of a currency as the concatenation of the
+		/// configured `GeneralCurrencyPrefix` and its local currency identifier.
+		///
+		/// Assumes the currency to be registered in the `AssetRegistry`.
+		pub fn try_get_general_index(
+			currency: <T as pallet::Config>::CurrencyId,
+		) -> Result<u128, DispatchError> {
+			ensure!(
+				<T as Config>::AssetRegistry::metadata(&currency).is_some(),
```

Comment:
I think there should be a way of calling `contains_key` instead of `metadata(_).is_some()` but this is ok 👍

Reply:
We restricted `AssetRegistry` to the `Inspect` trait which [only exposes](https://github.com/open-web3-stack/open-runtime-module-library/blob/16b6c1149a15674d21c87244b7988a667e2c14d9/traits/src/asset_registry.rs#L42) `fn metadata` for our needs.